@xen-orchestra/fs/src/s3.js

```diff
@@ -29,15 +31,29 @@ export default class S3Handler extends RemoteHandlerAbstract {
       },
     }
     if (protocol === 'http') {
-      params.httpOptions.agent = new http.Agent()
-      params.sslEnabled = false
+      throw new Error('HTTP no longer supported, please use HTTPS')
+    } else {
+      params.httpOptions.agent = new https.Agent({
+        // TODO : UI checkbox
+        rejectUnauthorized: false,
+      })
     }
     if (region !== undefined) {
       params.region = region
     }
 
     this._s3 = aws(params).s3
 
+    const [hostname, port] = host.split(':')
```

Comment:
Please a TODO/FIXME comment above, we should not do that as it's not compatible with IPv6.